api/src/main/java/org/apache/druid/indexer/TaskStatusPlus.java

```
@@ -42,28 +47,71 @@
+        statusCode,
+        statusCode,
+        runnerStatusCode,
+        duration,
+        location,
+        dataSource,
+        errorMsg
+    );
+  }
+
+
   @JsonCreator
   public TaskStatusPlus(
       @JsonProperty("id") String id,
       @JsonProperty("type") @Nullable String type, // nullable for backward compatibility
       @JsonProperty("createdTime") DateTime createdTime,
       @JsonProperty("queueInsertionTime") DateTime queueInsertionTime,
-      @JsonProperty("statusCode") @Nullable TaskState state,
+      @JsonProperty("statusCode") @Nullable TaskState statusCode,
+      @Deprecated @JsonProperty("status") @Nullable TaskState status,  // present for backwards compatibility
       @JsonProperty("runnerStatusCode") @Nullable RunnerTaskState runnerTaskState,
       @JsonProperty("duration") @Nullable Long duration,
       @JsonProperty("location") TaskLocation location,
       @JsonProperty("dataSource") @Nullable String dataSource, // nullable for backward compatibility
       @JsonProperty("errorMsg") @Nullable String errorMsg
   )
   {
-    if (state != null && state.isComplete()) {
+    if (statusCode != null && statusCode.isComplete()) {
       Preconditions.checkNotNull(duration, "duration");
     }
     this.id = Preconditions.checkNotNull(id, "id");
     this.type = type;
     this.createdTime = Preconditions.checkNotNull(createdTime, "createdTime");
     this.queueInsertionTime = Preconditions.checkNotNull(queueInsertionTime, "queueInsertionTime");
-    this.state = state;
+    //checks for deserialization safety
+    if (statusCode != null && status == null) {
+      this.statusCode = statusCode;
+      this.status = statusCode;
+    } else if (statusCode == null && status != null) {
+      this.statusCode = status;
+      this.status = status;
+    } else {
+      if (statusCode != null && status != null && statusCode != status) {
+        log.error("statusCode[%s] and status[%s] must be same", statusCode, status);
```

Is there any case that we can allow this mismatch? If so, it would be better `log.warn()`. Otherwise, it would be better to throw an exception.

---

Looks that this logging is now duplicate with the below exception. I think it's fine to remove it.

---

sure